### PR TITLE
added functionality : query the user's repositories and add them to the user settings on github connect

### PR DESCRIPTION
**Pull request by David-Muhasa**
…the user settings on github connect

### Issues

- [X] None

### Checklist

- [X] None

### How2Test

- [X] None

### Todo

- [X] None

### CODE REVIEW

**Comment by David-Muhasa** [comment]
added functionality : query the user's repositories and add them to the user settings on github connect
https://github.com/cypht-org/cypht/issues/332

File: modules/github/modules.php [hunk 1]
```diff
@@ -545,17 +561,32 @@ protected function output() {
...
+                '<input type="text" value="" class="form-control" placeholder="'.$this->trans('Owner').'" name="new_github_repo_owner" />'.
+                '<br><input type="submit" class="btn btn-secondary" name="github_add_repo" value="Add" />'.
+                '</form></div> <br>';
+            $res .= '<div class="configured_server"><h5><div class="server_title">'.$this->trans('Repositories').'</h5></div>';
             foreach ($this->get('github_repos', array()) as $repo) {
                 $res .= '<div class="configured_repo"><form class="remove_repo" method="POST">'.
                     '<input type="hidden" name="hm_page_key" value="'.$this->html_safe(Hm_Request_Key::generate()).'" />'.
                     '<input type="hidden" name="github_repo" value="'.$this->html_safe($repo).'" />'.
-                    '<input type="submit" name="github_remove_repo" value="'.$this->trans('Remove').'" class="github_remove_repo" />'.$this->html_safe($repo).'</form></div>';
+                    '<table class="table">
```
Comment (josaphatim) [inline modules/github/modules.php:574]: You should put this line, thead and body tags outside the loop

File: modules/github/modules.php [hunk 2]
```diff
@@ -630,13 +661,13 @@ protected function output() {
...
-            $res .= '<br /><form id="github_disconnect_form" method="POST">';
+            $res .='<h6>'. $this->trans('Already connected').'</h6>';
+            $res .= '<form id="github_disconnect_form" method="POST">';
             $res .= '<input type="hidden" name="hm_page_key" value="'.$this->html_safe(Hm_Request_Key::generate()).'" />';
-            $res .= '<input type="submit" name="github_disconnect" class="github_disconnect" value="'.$this->trans('Disconnect').'" />';
+            $res .= '<input type="submit" name="github_disconnect" class="github_disconnect btn btn-outline-danger" value="'.$this->trans('Disconnect').'" />';
             $res .= '</form>';
         }
-        return $res.'</div>';
+        return $res.'</div> <br>';
```
Comment (josaphatim) [inline modules/github/modules.php:670]: No need for adding br at the end

File: modules/github/modules.php [hunk 3]
```diff
@@ -630,13 +661,13 @@ protected function output() {
             $res .= '<a class="btn btn-secondary" href="'.$this->get('github_auth_url', '').'">'.$this->trans('Enable').'</a></div></div><div class="end_float"</div>';
         }
         else {
-            $res .= $this->trans('Already connected');
-            $res .= '<br /><form id="github_disconnect_form" method="POST">';
+            $res .='<h6>'. $this->trans('Already connected').'</h6>';
+            $res .= '<form id="github_disconnect_form" method="POST">';
```
Comment (josaphatim) [inline modules/github/modules.php:665]: col-12 col-lg-4 classes were added to other forms on the same page

File: lib/framework.php [hunk 4]
```diff
@@ -29,6 +29,7 @@
 require APP_PATH.'lib/crypt.php';
 require APP_PATH.'lib/crypt_sodium.php';
 require APP_PATH.'lib/sodium_compat.php';
+require APP_PATH.'lib/scram.php';
```
Comment (marclaporte) [inline lib/framework.php:32]: The diff mixes unrelated work. Please see https://en.wikipedia.org//wiki/Atomic_commit

**Comment by Yannick243** [comment]
Hello @josaphatim
After completing the authorization process on GitHub (during the callback), Cypht's authentication logs out, forcing the user to log in again. Have you encountered this behavior elsewhere or do you have any suggestions on how to resolve it ?

**Comment by josaphatim** [comment]
GMAIL_CLIENT_URI must point to ?page=home

for example: `GMAIL_CLIENT_URI=http://localhost/cypht/?page=home`

**Comment by Yannick243** [comment]
> GMAIL_CLIENT_URI must point to ?page=home
> 
> for example: `GMAIL_CLIENT_URI=http://localhost/cypht/?page=home`

This doesn't seem to work,
Any other idea ?

**Comment by josaphatim** [comment]
> > GMAIL_CLIENT_URI must point to ?page=home
> > for example: `GMAIL_CLIENT_URI=http://localhost/cypht/?page=home`
> 
> This doesn't seem to work, the user is still logged out Any other idea ?

Sorry it is `GITHUB_REDIRECT_URI=http://localhost/cypht/?page=home`

**Comment by Yannick243** [comment]
> GITHUB_REDIRECT_URI

This is what i used

**Pull request merged**